datafusion/common/src/logical_type.rs

```diff
@@ -0,0 +1,411 @@
+            Self::Struct(_) => "Struct",
+            Self::Map(_, _) => "Map",
+            Self::Decimal128(_, _) => "Decimal128",
+            Self::Decimal256(_, _) => "Decimal256",
+            Self::Extension(ext) => ext.display_name(),
+        }
+    }
+
+    fn type_signature(&self) -> TypeSignature {
+        match self {
+            Self::Boolean => TypeSignature::new("boolean"),
+            Self::Int32 => TypeSignature::new("int32"),
+            Self::Int64 => TypeSignature::new("int64"),
+            Self::UInt64 => TypeSignature::new("uint64"),
+            Self::Float32 => TypeSignature::new("float32"),
+            Self::Float64 => TypeSignature::new("float64"),
+            Self::String => TypeSignature::new("string"),
+            Self::Timestamp(tu, zone) => {
+                let tu = match tu {
+                    TimeUnit::Second => "second",
+                    TimeUnit::Millisecond => "millisecond",
+                    TimeUnit::Microsecond => "microsecond",
+                    TimeUnit::Nanosecond => "nanosecond",
+                };
+
+                let params = if let Some(zone) = zone {
+                    vec![tu.into(), zone.as_ref().into()]
+                } else {
+                    vec![tu.into()]
+                };
+
+                TypeSignature::new_with_params("timestamp", params)
+            }
+            Self::Binary => TypeSignature::new("binary"),
+            Self::Utf8 => TypeSignature::new("string"),
+            Self::Extension(ext) => ext.type_signature(),
+            Self::Struct(fields) => {
+                let params = fields.iter().map(|f| f.name().into()).collect();
+                TypeSignature::new_with_params("struct", params)
+            }
+            other => panic!("not implemented: {other:?}"),
+        }
+    }
+
+    fn physical_type(&self) -> DataType {
+        match self {
+            Self::Boolean => DataType::Boolean,
+            Self::Int32 => DataType::Int32,
+            Self::Int64 => DataType::Int64,
+            Self::UInt64 => DataType::UInt64,
+            Self::Float32 => DataType::Float32,
+            Self::Float64 => DataType::Float64,
+            Self::String => DataType::Utf8,
+            Self::Timestamp(tu, zone) => DataType::Timestamp(tu.clone(), zone.clone()),
+            Self::Binary => DataType::Binary,
+            Self::Utf8 => DataType::Utf8,
+            Self::Extension(ext) => ext.physical_type(),
+            Self::Struct(fields) => {
+                let fields = fields
+                    .iter()
+                    .map(|f| {
+                        let name = f.name();
+                        let data_type = f.physical_type();
+                        Arc::new(Field::new(name, data_type, true))
+                    })
+                    .collect::<Vec<_>>();
+                DataType::Struct(fields.into())
+            }
+            other => panic!("not implemented {other:?}"),
```

Comment:
is the idea that [`DataType::Dictionary`](https://docs.rs/arrow/latest/arrow/datatypes/enum.DataType.html#variant.Dictionary) and  [DataType::RunEndEncoded](https://docs.rs/arrow/latest/arrow/datatypes/enum.DataType.html#variant.RunEndEncoded) would also be included here? If so it makes sense

Reply:
Thanks for the heads up. This should include `DataType::Dictionary` and `DataType::RunEndEncoded` here.